plugins/extractors/clickhouse/clickhouse.go

```diff
@@ -53,13 +55,27 @@ func New(logger log.Logger) *Extractor {
 }
 
 // Init initializes the extractor
-func (e *Extractor) Init(ctx context.Context, config plugins.Config) error {
-	if err := e.BaseExtractor.Init(ctx, config); err != nil {
+func (e *Extractor) Init(ctx context.Context, config plugins.Config) (err error) {
+	err = e.BaseExtractor.Init(ctx, config)
+	if err != nil {
 		return err
 	}
 
-	var err error
-	e.db, err = sql.Open("clickhouse", e.config.ConnectionURL)
+	driverName := "clickhouse"
+	if config.OtelEnabled {
+		driverName, err = otelsql.Register(driverName,
+			otelsql.AllowRoot(),
```

Comment:
https://github.com/nhatthm/otelsql#allowroot-and-span-context

> To fully take advantage of otelsql, all database calls should be made using the *Context methods. Failing to do so will result in many orphaned traces if the AllowRoot() is used. By default, AllowRoot() is disabled and will result in otelsql not tracing the database calls if context or parent spans are missing.

Lets not have `AllowRoot()` enabled. We should modify the queries to take the context instead.